### PR TITLE
modules/SceSysmem: stub sceKernelOpenMemBlock.

### PR DESCRIPTION
**Pull request by Zangetsu38**

### CODE REVIEW

File: vita3k/modules/SceSysmem/SceSysmem.cpp [hunk 1]
```diff
@@ -62,6 +62,7 @@ struct SysmemState {
     std::mutex mutex;
     Blocks blocks;
     Blocks vm_blocks;
+    std::map<SceUID, char[KERNELOBJECT_MAX_NAME_LENGTH + 1]> mem_block_names;
```
Comment (Macdu) [inline vita3k/modules/SceSysmem/SceSysmem.cpp:65]: No need to add a new map for this

Comment (Zangetsu38) [reply vita3k/modules/SceSysmem/SceSysmem.cpp:65]: In first i have added it inside BlockInfo, but after check in henckaku blockinfo no have name, but is inside other structure called  BlockInfoDetail
Ask @bookmist

Comment (Macdu) [reply vita3k/modules/SceSysmem/SceSysmem.cpp:65]: Just find a way to add it to block while keeping the MemBlockInfo structure the same. They are many different ways to do this (inheritance, creating a structure containing the block info and the name mainly)

Comment (Zangetsu38) [reply vita3k/modules/SceSysmem/SceSysmem.cpp:65]: ok i no see i have no idea how done this

File: vita3k/modules/SceSysmem/SceSysmem.cpp [hunk 2]
```diff
@@ -301,9 +308,21 @@ EXPORT(bool, sceKernelIsPSVitaTV) {
...
+    const std::lock_guard<std::mutex> memblock_lock(state->mutex);
+
+    const auto it = std::find_if(state->mem_block_names.begin(), state->mem_block_names.end(), [=](auto it) {
+        return strncmp(it.second, name, KERNELOBJECT_MAX_NAME_LENGTH) == 0;
+    });
+
+    if (it != state->mem_block_names.end())
+        return it->first;
+
+    return RET_ERROR(SCE_KERNEL_ERROR_ILLEGAL_BLOCK_ID);
```
Comment (Macdu) [inline vita3k/modules/SceSysmem/SceSysmem.cpp:325]: Error is not correct

Comment (Zangetsu38) [reply vita3k/modules/SceSysmem/SceSysmem.cpp:325]: Hmm probably yes

File: vita3k/modules/SceSysmem/SceSysmem.cpp (hunk 1 above)
Comment (Zangetsu38) [reply vita3k/modules/SceSysmem/SceSysmem.cpp]: is done

File: vita3k/modules/SceSysmem/SceSysmem.cpp (hunk 2 above)
Comment (Zangetsu38) [reply vita3k/modules/SceSysmem/SceSysmem.cpp]: done too

File: vita3k/modules/SceSysmem/SceSysmem.cpp [hunk 3]
```diff
@@ -55,6 +55,10 @@ struct SceKernelFreeMemorySizeInfo {
     int size_phycont; //!< Free memory size for USER_MAIN_PHYCONT_*_RW memory
 };
 
+struct KernelMemBlock : SceKernelMemBlockInfo {
+    char name[KERNELOBJECT_MAX_NAME_LENGTH + 1];
+};
+
 typedef std::shared_ptr<SceKernelMemBlockInfo> SceKernelMemBlockInfoPtr;
 typedef std::map<SceUID, SceKernelMemBlockInfoPtr> Blocks;
```
Comment (Macdu) [inline vita3k/modules/SceSysmem/SceSysmem.cpp:63]:
```suggestion
typedef std::shared_ptr<KernelMemBlock> KernelMemBlockPtr;
typedef std::map<SceUID, KernelMemBlockPtr> Blocks;
```

Comment (Zangetsu38) [reply vita3k/modules/SceSysmem/SceSysmem.cpp]: Done, i have also updated so std::;find no need reinterpret now and works fine 👌

**Review by Macdu — APPROVED** [review]

**Pull request merged**